src/main/java/chessmaster/game/ChessBoard.java

```diff
@@ -395,8 +396,9 @@ public boolean isEndGame() {
     }
 
     public Color getWinningColor() {
-        
-        if (isCheckmated(playerColor)) {
+        if (isCheckmated(playerColor) && isCheckmated(playerColor.getOppositeColour())) {
+            return Color.DRAW;
```

Comment:
So it only draws the game if _both_ players are checkmated at the same time? Will this condition ever be true? To be honest I'm not entirely sure how you would define stalemate – I thought it was when one player isn't in checkmate but can't make any moves.

Reply:
Here, the isCheckmated function searches for any available moves. A stalemate can also occur when there are no available moves for either party, even though neither might be technically in check.

Reply:
Sounds good.